internal/handlers/vault_handler.go

```diff
@@ -59,15 +60,88 @@ func (a *Api) getVaultHandler(c *gin.Context) {
 		return
 	}
 	vaultResp := models.VaultResponse{
+		UId:            vault.Uid,
 		Name:           vault.Name,
+		Alias:          vault.Alias,
 		PublicKeyECDSA: vault.ECDSA,
 		PublicKeyEDDSA: vault.EDDSA,
 		TotalPoints:    vault.TotalPoints,
 		JoinAirdrop:    vault.JoinAirdrop,
-		Coins:          []models.CoinDBModel{},
+		Coins:          []models.ChainCoins{},
 	}
 	for _, coin := range coins {
-		vaultResp.Coins = append(vaultResp.Coins, coin)
+		found := false
+		for i, _ := range vaultResp.Coins {
+			if vaultResp.Coins[i].Name == coin.Chain {
+				vaultResp.Coins[i].Coins = append(vaultResp.Coins[i].Coins, models.NewCoin(coin))
+				found = true
+			}
+		}
+		if !found {
+			vaultResp.Coins = append(vaultResp.Coins, models.ChainCoins{
+				Name:         coin.Chain,
+				Address:      coin.Address,
+				HexPublicKey: coin.HexPublicKey,
+				Coins:        []models.Coin{models.NewCoin(coin)},
+			})
+		}
+	}
+	c.JSON(http.StatusOK, vaultResp)
+}
+
+func (a *Api) getVaultByUIDHandler(c *gin.Context) {
+	uid := c.Param("uid")
+	if uid == "" {
+		c.Error(errInvalidRequest)
+		return
+	}
+	vault, err := a.s.GetVaultByUID(uid)
+	if err != nil {
+		a.logger.Error(err)
+		c.Error(errFailedToGetVault)
+		return
+	}
+	if vault == nil {
+		c.Error(errVaultNotFound)
+		return
+	}
+	coins, err := a.s.GetCoins(vault.ID)
+	if err != nil {
+		a.logger.Error(err)
+		c.Error(errFailedToGetCoin)
+		return
+	}
+	if vault.Alias == "" {
+		vault.Alias = vault.Name
+	}
+	vaultResp := models.VaultResponse{
+		UId:            vault.Uid,
+		Name:           vault.Alias,
+		PublicKeyECDSA: "",
```

Comment:
why you leave these two public keys empty?  I think it is ok to return these two